src/app/store/chats-store/chats.selectors.ts

```diff
@@ -0,0 +1,9 @@
+import { createFeatureSelector, createSelector } from '@ngrx/store';
+import { ChatState, chatFeatureKey } from './chats.reducer';
+
+const selectMessagesState = createFeatureSelector<ChatState>(chatFeatureKey);
+
+export const selectMessages = createSelector(
+  selectMessagesState,
+  (state) => state.messages
```

Comment:
Do your transforms to get time and truncate user here as you map over each message